tests/sparse-set.cc

```diff
@@ -0,0 +1,116 @@
+//
+// Created by Lukáš Holík on 06.12.2022.
+//
+
+#include "../3rdparty/catch.hpp"
+
+#include "mata/nfa.hh"
+#include "mata/sparse-set.hh"
+
+using namespace Mata::Util;
+using namespace Mata::Nfa;
+
+TEST_CASE("Mata::Util::SparseSet") {
+    SparseSet<State> p;
+
+    SECTION("basic functionality: insert, erase, access, constructor, size, get_elements ") {
+        std::vector<State> v = {1, 2, 3, 4, 5};
+        p.insert_all(v);
```

Comment:
I will have a look at the `insert_all` function and try to replace it with a templated `insert` as was originally intended in the near future.